src/__swaps__/utils/swaps.ts

```diff
@@ -295,37 +291,59 @@ export function niceIncrementFormatter({
   percentageToSwap: number;
   sliderXPosition: number;
   stripSeparators?: boolean;
+  isStablecoin?: boolean;
 }) {
   'worklet';
+
+  function precisionBasedOffMagnitude(amount: number | string): number {
+    const magnitude = -Number(floorWorklet(sumWorklet(log10Worklet(amount), 1)));
+    // don't let stablecoins go beneath 2nd order
+    if (magnitude < -2 && isStablecoin) {
+      return -2;
+    }
+
+    return magnitude;
+  }
+
   if (percentageToSwap === 0) return '0';
-  if (percentageToSwap === 0.25)
+  if (percentageToSwap === 0.25) {
+    const amount = mulWorklet(inputAssetBalance, 0.25);
     return valueBasedDecimalFormatter({
-      amount: mulWorklet(inputAssetBalance, 0.25),
+      amount,
       usdTokenPrice: inputAssetUsdPrice,
       roundingMode: 'up',
-      precisionAdjustment: -3,
+      precisionAdjustment: precisionBasedOffMagnitude(amount),
+      isStablecoin,
     });
-  if (percentageToSwap === 0.5)
+  }
+  if (percentageToSwap === 0.5) {
+    const amount = mulWorklet(inputAssetBalance, 0.5);
     return valueBasedDecimalFormatter({
-      amount: mulWorklet(inputAssetBalance, 0.5),
+      amount,
       usdTokenPrice: inputAssetUsdPrice,
       roundingMode: 'up',
-      precisionAdjustment: -3,
+      precisionAdjustment: precisionBasedOffMagnitude(amount),
+      isStablecoin,
     });
-  if (percentageToSwap === 0.75)
+  }
+  if (percentageToSwap === 0.75) {
+    const amount = mulWorklet(inputAssetBalance, 0.75);
     return valueBasedDecimalFormatter({
-      amount: mulWorklet(inputAssetBalance, 0.75),
+      amount,
       usdTokenPrice: inputAssetUsdPrice,
       roundingMode: 'up',
-      precisionAdjustment: -3,
+      precisionAdjustment: precisionBasedOffMagnitude(amount),
+      isStablecoin,
     });
-  if (percentageToSwap === 1)
+  }
+  if (percentageToSwap === 1) {
     return valueBasedDecimalFormatter({
       amount: inputAssetBalance,
       usdTokenPrice: inputAssetUsdPrice,
-      roundingMode: 'up',
```

Comment:
if we're at 100% percent we want to round normally, not exceed their balance

Reply:
Good catch!